modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/menu.js

```diff
@@ -465,14 +465,8 @@ AUI.add(
 					'aria-haspopup': true,
 				});
 
-				listNode.setAttribute('aria-labelledby', trigger.guid());
-
-				if (
-					trigger._stateProxy?.classList?.contains(
-						'input-localized-trigger'
-					)
-				) {
-					listNode.removeAttribute('aria-labelledby', trigger.guid());
+				if (!trigger.hasClass('input-localized-trigger')) {
```

Comment:
That's a similar scenario as the other commit, we rely on implicit information (trigger css class) to determine behavior (setting aria attribute).

In the end, it seems that the treatment we want to give the icon-menu when rendered from the input-localized is a bit different than the rest of the cases.

We might want to differentiate this singularity with a single mechanism, even if it's not made clear in the tag api @markocikos makes sense?

Reply:
It's not terrible to determine logic based on CSS class, specially as that class is also related to localization. I would tend to be more lenient to solution, as it is on a legacy AUI component. In AUI, DOM manipulation is commonplace. But I agree it's not ideal. I am not sure what mechanism you have in mind